File: CPA6_functions.java
public class CPA6_functions{
    String  n;
    public CPA6_functions(String  n){
        this.n = n;
    }
    
    public int find_sum(String n){
        try {
            int sum_num = 0;
            int i = 0;
            while(i < this.n.length()){

                char num = this.n.charAt(i);
                String number = Character.toString(num);// converts char to string
                int number1 = Integer.parseInt(number);
                System.out.println(number1);
                sum_num  = sum_num + number1;
                i++;
            }
            System.out.println(sum_num);
            return sum_num;
            }

        catch (Exception e) {
            System.out.println("Please check your input,it should be a whole number");
        }
        return 0;
    }

    }

File: CPA6_functions.py
'''Author: Khadijatu Nayi Alhassan
 CPA6'''


def find_sum(n):
    '''This function takes find the sum of the digits that make up a number.
    paramters: 
        n: a whole number
    returns a sum of the digits in n'''
    
    try:
        ''' This part of the code deals with handling errors. It prints a statement
        to guide the user to enter the right input if he enters a decimal number,
        a negative number or a void input'''
        sum_num = 0
        let = str(n)

        for i in let:
            num = int(i)
            sum_num +=num
        return sum_num

    except SyntaxError:
        print('please enter a whole number')
    except ValueError:
        print("your input contains a character other than an integer")
    except TypeError:
        print('please enter a number, your input is void')


    



